odoo/addons/base/tests/test_res_country.py

```diff
@@ -1,21 +1,63 @@
 from odoo.tests import TransactionCase, tagged
 
 
-@tagged('-at_install', 'post_install')
-class TestResCountryState(TransactionCase):
-    def test_find_by_name(self):
-        """It should be possible to find a state by its display name
-        """
-        glorious_arstotzka = self.env['res.country'].create({
+class TestResCountryCommon(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.glorious_arstotzka = cls.env['res.country'].create({
             'name': 'Arstotzka',
             'code': 'AA',
         })
-        altan = self.env['res.country.state'].create({
-            'country_id': glorious_arstotzka.id,
+        cls.altan = cls.env['res.country.state'].create({
+            'country_id': cls.glorious_arstotzka.id,
             'code': 'AL',
             'name': 'Altan',
         })
 
+
+@tagged('-at_install', 'post_install')
+class TestResCountry(TestResCountryCommon):
+    def test_name_search(self):
+        glorious_arstotzka_tuple = (self.glorious_arstotzka.id, self.glorious_arstotzka.display_name)
+        for (name, op) in [
+            ('ARSTOTZKA', '='),
+            ('arstotzka', '='),
+            ('ARSTOTZKA', '!='),
+            ('arstotzka', '!='),
+            (['ARSTOTZKA'], 'in'),
+            (['arstotzka'], 'in'),
+            (['ARSTOTZKA'], 'not in'),
+            (['arstotzka'], 'not in'),
+        ]:
+            with self.subTest((name, op)):
+                if op in ('!=', 'not in'):
+                    self.assertFalse(
```

Comment:
`assertFunc = self.assertNotIn if op in ('!=', 'not in') else self.assertEqual`?